File: client/src/app/models/league.ts
import { UserTeam, Game,LeagueSettings } from ".";

export type League = {
    id : number
    name : string
    teams : UserTeam[]
    schedule : Game[]
    season : number
    settings:LeagueSettings
    adminId : string
}

export type CreateLeagueDto = {
    name : string
}

File: Core/Entities/League.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Core.Entities;

public class League : BaseEntity
{
    public required string Name { get; set; }
    public IList<UserTeam> Teams { get; set; } = [];
    public required string AdminId { get; set; }
    [NotMapped] public IList<Game> Schedule { get; set; } = [];
    public required int Season { get; set; }
    public int CurrentWeek { get; set; }
    [NotMapped] public LeagueSettings Settings { get; set; } = null!;

    public static League CreateLeague(string name, int season, string adminId)
    {
        return new League
        {
            Name = name,
            Season = season,
            AdminId = adminId
        };
    }
}